### PR TITLE
feat(groups): Customer usage with groups

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  https://github.com/getlago/lago/issues/32

## Context

We want to be able to price differently billable metrics according to several dimensions (groups).

For example, creating a `storage` pricing based on the `region of the server` and the `cloud provider` used.

## Description

The goal of this PR is to be able to support groups on customer current usage.

### CODE REVIEW

**Review by vincent-pochet — APPROVED** [review]
:rocket:

File: spec/graphql/resolvers/customers/usage_resolver_spec.rb [hunk 1]
```diff
@@ -101,12 +102,180 @@
...
+        )
+      end
+    end
+  end
+
+  context 'with two dimensions group' do
+    let(:aws) { create(:group, billable_metric: metric, key: 'cloud', value: 'aws') }
+    let(:google) { create(:group, billable_metric: metric, key: 'cloud', value: 'google') }
+    let(:aws_usa) { create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: aws.id) }
+    let(:aws_france) { create(:group, billable_metric: metric, key: 'region', value: 'france', parent_group_id: aws.id) }
```
Comment (jdenquin) [inline spec/graphql/resolvers/customers/usage_resolver_spec.rb:192]: Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)

File: spec/graphql/resolvers/customers/usage_resolver_spec.rb [hunk 2]
```diff
@@ -101,12 +102,180 @@
...
+    let(:google_usa) { create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: google.id) }
```
Comment (jdenquin) [inline spec/graphql/resolvers/customers/usage_resolver_spec.rb:193]: Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)

File: spec/requests/api/v1/customers_spec.rb [hunk 3]
```diff
@@ -131,13 +131,172 @@
...
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'google', region: 'usa' },
+        )
+      end
+
+      it 'returns the group usage for the customer' do
+        get_with_token(
+          organization,
+          "/api/v1/customers/#{customer.external_id}/current_usage?external_subscription_id=#{subscription.external_id}",
```
Comment (jdenquin) [inline spec/requests/api/v1/customers_spec.rb:283]: Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)

**Pull request merged**